filebeat/harvester/source/source.go

```diff
@@ -15,3 +17,12 @@ type FileSource interface {
 	Stat() (os.FileInfo, error)
 	Continuable() bool // can we continue processing after EOF?
 }
+
+func NewFile(file *os.File, compression string) (FileSource, error) {
+
+	if config.GZipCompression == compression {
```

Comment:
Overall I see multiple places in code with logic specific to GZip. When new compression type is supported, we'd need to modify all these places.. Is it possible to have all logic per compression type encapsulated in a single place?